lib/ansible/module_utils/facts.py

```diff
@@ -515,6 +521,15 @@ def get_pkg_mgr_facts(self):
         if self.facts['system'] == 'OpenBSD':
                 self.facts['pkg_mgr'] = 'openbsd_pkg'
 
+    def get_service_mrg_facts(self):
+        self.facts['service_mgr'] = 'auto'
+        for service in Facts.INIT_SYSTEM_MGRS:
```

Comment:
this is not a good way to detect initd systems, upstart, openrc, rc systems will all be confused into init